src/info/repo.rs

```diff
@@ -155,7 +155,7 @@ impl Commits {
 
 impl Repo {
     pub fn new(git2_repo: Repository) -> Result<Self> {
-        let repo = git::open(git2_repo.path())?;
+        let repo = git::open(git2_repo.workdir().unwrap_or(git2_repo.path()))?;
```

Comment:
As a quick double-check, looks like `.workdir()` only returns `None` if the repository is bare. So effectively this attempts to open either the workdir path or the path to the bare repository, right? Just checking that that's wanted.

Reply:
That's a great catch! Actually this was a workaround to `gitoxide` being unable to open `.git/worktrees/<name>` paths at the time, which has since been fixed. So I rolled it back to `git2_repo.path()` exactly like before.

As a side-note, while implementing worktree support it became clear that I have to change my thinking about what `bare` really means. In short: don't assume that `bare` repositories have no worktree. In fact, they can have several _linked_ worktrees via `git worktree`. The only way to know if there is a worktree is to open a repository and checking for the presence of a `work_dir()`. Oh, to make things more fun: this directory might not even exist as it's just something pointed to by a file that we didn't have to touch yet (and which is exactly what can happen if a repo is opened through `.git/worktrees/<name>`.